google/cloud/storage/blob.py

```diff
@@ -2783,6 +2782,11 @@ def create_resumable_upload_session(
         client=None,
         timeout=_DEFAULT_TIMEOUT,
         checksum=None,
+        if_generation_match=None,
```

Comment:
Adding these parameters here seems kind of separate from the rest of these changes. Is this method called under the hood by other retry operations, or just directly by external callers?

Reply:
`create_resumable_upload_session()` is called directly by external callers. This is to add preconditions and retry configuration. I think it may make more sense to separate this out as an independent PR. Thanks @tritone and @andrewsg for the input!

Reply:
Done, thanks! Split into PR #484